vllm/worker/model_runner.py

```diff
@@ -646,82 +655,94 @@ def _prepare_model_input(
-            sampling_metadata = SamplingMetadata.prepare(
-                seq_group_metadata_list, seq_lens, query_lens, self.device,
-                self.pin_memory)
-
-            metadata_dict = {
-                "input_tokens": input_tokens,
-                "input_positions": input_positions,
-                "selected_token_indices":
-                sampling_metadata.selected_token_indices,
-                "lora_requests": lora_requests,
-                "lora_mapping": lora_mapping,
-                "multi_modal_kwargs": multi_modal_kwargs,
-                "num_prefill_tokens": num_prefill_tokens,
-                "num_decode_tokens": num_decode_tokens,
-                "slot_mapping": slot_mapping,
-                "num_prefills": num_prefills,
-            }
-            if attn_metadata:
-                metadata_dict.update(attn_metadata.asdict_zerocopy())
-            broadcast_tensor_dict(metadata_dict, src=0)
+    def prepare_inputs_to_broadcast(
+        self, seq_group_metadata_list: List[SequenceGroupMetadata]
+    ) -> Tuple[Dict, Optional[List]]:
+        """
+        This function prepares the input tensors for the model. The output
+        is a dictionary that will be broadcasted to all workers, and a list
+        of auxiliary data that will be passed to the model execution function
+        for the driver worker.
+        """
+        # Prepare input tensors.
+        (
+            input_tokens,
+            input_positions,
+            attn_metadata,
+            seq_lens,
+            query_lens,
+            lora_mapping,
+            lora_requests,
+            multi_modal_kwargs,
+            slot_mapping,
+            num_prefill_tokens,
+            num_decode_tokens,
+            num_prefills,
+        ) = self._prepare_model_input(seq_group_metadata_list)
+        sampling_metadata = SamplingMetadata.prepare(seq_group_metadata_list,
+                                                     seq_lens, query_lens,
+                                                     self.device,
+                                                     self.pin_memory)
+
+        metadata_dict = {
+            "input_tokens": input_tokens,
+            "input_positions": input_positions,
+            "selected_token_indices": sampling_metadata.selected_token_indices,
+            "lora_requests": lora_requests,
+            "lora_mapping": lora_mapping,
+            "multi_modal_kwargs": multi_modal_kwargs,
+            "num_prefill_tokens": num_prefill_tokens,
+            "num_decode_tokens": num_decode_tokens,
+            "slot_mapping": slot_mapping,
+            "num_prefills": num_prefills,
+        }
+        if attn_metadata:
+            metadata_dict.update(attn_metadata.asdict_zerocopy())
+
+        return metadata_dict, [sampling_metadata]
+
+    def convert_broadcast_inputs_to_modelrunner_input(
+            self, metadata_dict: Dict[str, Any], aux: Optional[List[Any]]):
```

Comment:
Add return typing?

Reply:
I delete it to avoid typing conflict. This function `convert_broadcast_inputs_to_modelrunner_input` is implemented in `embedding_model_runner.py` as well, but the return types differ (although both are named `ModelRunnerInput`, they live in different modules, so mypy complains about it).

Suggestions are welcome for fixing it while making mypy happy.